### PR TITLE
Don't require dplyr anywhere

### PR DESCRIPTION
**Pull request by lorenzwalthert**
Superseeds #1128.

### CODE REVIEW

File: tests/testthat/test-cache-low-level-api.R [hunk 1]
```diff
@@ -1,16 +1,16 @@
 test_that("caching utils make right blocks with semi-colon", {
   blocks_simple_uncached <- compute_parse_data_nested(c("1 + 1", "2; 1+1")) %>%
-    dplyr::mutate(is_cached = FALSE) %>%
+    mutate(is_cached = FALSE) %>%
```
Comment (MichaelChirico) [inline tests/testthat/test-cache-low-level-api.R:3]: where's `%>%` coming from if not dplyr?

Comment (MichaelChirico) [reply tests/testthat/test-cache-low-level-api.R:3]: nvm :)

File: R/compat-dplyr.R [hunk 2]
```diff
@@ -57,3 +57,12 @@ map_dfr <- function(.x, .f, ...) {
   res <- map(.x, .f, ...)
   vec_rbind(!!!res)
 }
+
+mutate <- function(.data, ...) {
```
Comment (MichaelChirico) [inline R/compat-dplyr.R:61]: use base::transform() instead? use case here seems like a drop-in

File: tests/testthat/test-cache-low-level-api.R (hunk 1 above)
Comment (lorenzwalthert) [reply tests/testthat/test-cache-low-level-api.R:3]: {magritrr}? We'll at some  point transition to `|>` if minimal supported R version has it.

File: R/compat-dplyr.R (hunk 2 above)
Comment (lorenzwalthert) [reply R/compat-dplyr.R:61]: ok sure.

File: vignettes/customizing_styler.Rmd [hunk 3]
```diff
@@ -50,17 +50,17 @@ As the name says, this function removes spaces after the opening parenthesis. Bu
 string_to_format <- "call( 3)"
 pd <- styler:::compute_parse_data_nested(string_to_format) %>%
   styler:::pre_visit_one(default_style_guide_attributes)
-pd$child[[1]] %>%
-  select(token, terminal, text, newlines, spaces)
+
+cols <- c('token', 'terminal', 'text', 'newlines', 'spaces')
+pd$child[[1]][, cols]
```
Comment (MichaelChirico) [inline vignettes/customizing_styler.Rmd:55]: maybe `subset(pd$child[[1]], select = cols)`?

Comment (MichaelChirico) [reply vignettes/customizing_styler.Rmd:55]: or since we've magrittr:

```r
pd$child[[1]] %>%
  subset(select = c(...))
```

**Review by MichaelChirico — APPROVED** [review]
LGTM! Comment is optional/up to preference.

File: vignettes/customizing_styler.Rmd (hunk 3 above)
Comment (lorenzwalthert) [reply vignettes/customizing_styler.Rmd:55]: I don't know... the audience of the vignette is probably R developers who know `subset()` but I think it's one of these functions that are not used often anymore these days... `]` is not very pretty here I know but probably easier to understand for most.

**Pull request merged**